[Resilient HTTP apps] Add a paragraph describing Http.Resilience RemoveAllResilienceHandlers method (#45335)
dotnet/docs: docs/core/resilience/http-resilience.md

There are several resilience-centric extensions available. Some are standard, thus employing various industry best practices, and others are more customizable. When adding resilience, you should only add one resilience handler and avoid stacking handlers. If you need to add multiple resilience handlers, you should consider using the `AddResilienceHandler` extension method, which allows you to customize the resilience strategies.

> [!IMPORTANT]

> > All the examples within this article rely on the <xref:Microsoft.Extensions.DependencyInjection.HttpClientFactoryServiceCollectionExtensions.AddHttpClient%2A> API, from the [Microsoft.Extensions.Http](https://www.nuget.org/packages/Microsoft.Extensions.Http) library, which returns an <xref:Microsoft.Extensions.DependencyInjection.IHttpClientBuilder> instance. The <xref:Microsoft.Extensions.DependencyInjection.IHttpClientBuilder> instance is used to configure the <xref:System.Net.Http.HttpClient> and add the resilience handler.

Reviewer: If you remove of, you also need to remove the. "All of the examples", or "All examples" are correct.

```suggestion
> All examples within this article rely on the <xref:Microsoft.Extensions.DependencyInjection.HttpClientFactoryServiceCollectionExtensions.AddHttpClient%2A> API, from the [Microsoft.Extensions.Http](https://www.nuget.org/packages/Microsoft.Extensions.Http) library, which returns an <xref:Microsoft.Extensions.DependencyInjection.IHttpClientBuilder> instance. The <xref:Microsoft.Extensions.DependencyInjection.IHttpClientBuilder> instance is used to configure the <xref:System.Net.Http.HttpClient> and add the resilience handler.
```

The preceding code adds the standard resilience handler to the <xref:System.Net.Http.HttpClient>. Like most resilience APIs, there are overloads that allow you to customize the default options and applied resilience strategies.

## Remove standard resilience handler

> 

Reviewer 2: Could you, please, add some intro? Something like this:

"We provide a method <xref:RemoveAllResilienceHandlers> which removes all previously registered resilience handlers. It is useful when you need to clear all existing resilience handlers and add your custom one."

The following example demonstrates how to configure a custom <xref:System.Net.Http.HttpClient> using the `AddHttpClient` method, remove all predefined resilience strategies, and add new custom ones. 
This approach allows you to clear existing configurations and define new ones according to your specific requirements.
:::code language="csharp" source="snippets/http-resilience/Program.RemoveHandlers.cs" id="remove-handlers":::

The preceding code:

- Adds the standard resilience handler to the named <xref:System.Net.Http.HttpClient> instance

> - Removes all predefined resilience handlers that were previously registered. This is useful when you want to start with a clean slate and add your own custom strategies.

Reviewer 2: ```suggestion
- Removes all predefined resilience handlers that were previously registered. This is useful when you want to start with a clean state and add your own custom strategies.
```

> - Adds a `StandardHedgingHandler` to the <xref:System.Net.Http.HttpClient>.You can replace `AddStandardHedgingHandler()` with any strategy that suits your application's needs, such as retry mechanisms, circuit breakers, or other resilience techniques.

Reviewer 2: ```suggestion
- Adds a `StandardHedgingHandler` to the <xref:System.Net.Http.HttpClient>. You can replace `AddStandardHedgingHandler()` with any strategy that suits your application's needs, such as retry mechanisms, circuit breakers, or other resilience techniques.
```

The preceding code adds the standard resilience handler to the <xref:System.Net.Http.HttpClient>. Like most resilience APIs, there are overloads that allow you to customize the default options and applied resilience strategies.

## Remove standard resilience handlers

> We provide a method <xref:Microsoft.Extensions.DependencyInjection.ResilienceHttpClientBuilderExtensions.RemoveAllResilienceHandlers*> which removes all previously registered resilience handlers. It is useful when you need to clear all existing resilience handlers and add your custom one.

Reviewer: ```suggestion
We provide a method <xref:Microsoft.Extensions.DependencyInjection.ResilienceHttpClientBuilderExtensions.RemoveAllResilienceHandlers*> which removes all previously registered resilience handlers. It is useful when you need to clear existing resilience handlers to add your custom one.
```

Author: Done as suggested

The following example demonstrates how to configure a custom <xref:System.Net.Http.HttpClient> using the `AddHttpClient` method, remove all predefined resilience strategies, and replace them with new handlers.
This approach allows you to clear existing configurations and define new ones according to your specific requirements.

:::code language="csharp" source="snippets/http-resilience/Program.RemoveHandlers.cs" id="remove-handlers":::

The preceding code:

- Creates a <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> instance.

> - Adds the standard resilience handler to the named <xref:System.Net.Http.HttpClient> instance.

Reviewer 2: ```suggestion
- Adds the standard resilience handler to all <xref:System.Net.Http.HttpClient> instances.
```

If this item refers to the line of code where you invoke `ConfigureHttpClientDefaults` then I think it will be clearer to write that we register the resilience handler for HttpClient instances.

> - Removes all predefined resilience handlers that were previously registered. This is useful when you want to start with a clean state and add your own custom strategies.

Reviewer: ```suggestion
- Removes all predefined resilience handlers that were previously registered. This is useful when you want to start with a clean state to add your own custom strategies.
```

Author: Done as suggested

> - Adds a `StandardHedgingHandler` to the <xref:System.Net.Http.HttpClient>. You can replace `AddStandardHedgingHandler()` with any strategy that suits your application's needs, such as retry mechanisms, circuit breakers, or other resilience techniques.

Reviewer 3: We may want to add a second indentation to this two list elements, since they refer to the previous one
